src/DotNetWorker.Core/DotNetWorker.Core.csproj

```diff
@@ -22,6 +21,16 @@
 
   <ItemGroup>
     <PackageReference Include="Azure.Core" Version="1.41.0" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'net9.0'">
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="9.0.0-preview.*" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="9.0.0-preview.*" />
+    <PackageReference Include="System.Collections.Immutable" Version="9.0.0-preview.*" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="9.0.0-preview.*" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'net9.0'">
     <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
```

Comment:
As part of #1095 we should discuss we need to have 8x,7x and 6x version of these dependencies updated  similar to what we are doing here for net9.